### DISCUSSION TITLE
Windows and RaspberryPi problems.

### DISCUSSION THREAD

**Discussion by Dhextreme** [GerryDazoo/Slinger#122] [Q&A] [answered]
I there.
I'm so glad to see some interest for the Slingbox.
I use it every days and it's the only way to watch tv for us now so I've installed Slinger on my Win10 and it's running great except for the fact that I can't use Vlc or an other viewer on the same computer where the server is installed.
I get a WinError 10053 but it work fine elsewhere.
If you have an idea of what's append here?

Also, I've tried to run it on my RaspberryPi but I'm not a crack of that system.
I followed instructions from the read me file as best that I can.
Look like it's installed but it's not running because I can't connect to it so if someone have an idea of my problem,
I want to figure out what's my problem.

Thanks GerryDazoo, you rock!!

**Comment by GerryDazoo** [comment]
I do this everyday. What URL are you giving to VLC? Is your local firewall blocking the requests?
Try ip address 127.0.0.1  this should work

**Reply by Dhextreme** [reply]
127.0.0.1 is what I use. 
I have followed Osymoko ''Multiple Servers issue'' thread also with no success.
I'll try it on my other windows to see what's append there, it's a fresh install.

**Comment by waldorf-statler** [comment]
Just wanted to mention that I have what seems like the same problem (for windows) - the stream is terminated as soon as it gets started if the vlc app is on the same machine as the code / server. It works fine everywhere else (IPAD, Firestick, other windows machine, etc...) with server code running on that pc. Have tried playing with firewall and AV to no avail. No big deal but just in case it's the same issue as Dhextreme, and/or it is obvious to anyone. Get the following..

 Starting Stream for  192.168.0.29:52805
Connecting... ('192.168.0.31', 5324) Control
VideoParameters: Resolution= 12 FrameRate= 30 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 64
Connecting... ('192.168.0.31', 5324) Stream
 Stream started at 10/10/2022, 18:55:34  744 0
10/10/2022, 18:55:38   Stream Terminated for  192.168.0.29:52805 [WinError 10053] An established connection was aborted by the software in your host machine
 Shutting down connections
Streamer:   Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox

**Reply by GerryDazoo** [reply]
Have a look at you VLC error log to see why it's shutting down the connection. I can stream all day long without any issues.

**Reply by Dhextreme** [reply]
I did that GerryDazoo and this is the log.

-- logger module started --
main: Lancement de vlc avec l'interface par défaut. Utiliser « cvlc » pour démarrer VLC sans interface.
main: playlist is empty
mjpeg warning: no more data
mjpeg error: cannot peek
avcodec warning: cannot peek
-- logger module stopped --

**Comment by Dhextreme** [comment]
Ok. After some testing on my fresh Win 10 machine, it 's working perfectly both server and Vlc so the problem come from my other Windows.
I'll not go further here.
I'll figure out my Win problem, maybe a conflict with an other software so thanks a lot for this part!!

Now I can say VLC is not working between my everyday pc and the server installed on my fresh win 10, 
but the server work fine on my everyday pc and viewing from any other viewer on other pc, phone or tablet so that confirm that is not the server but something in communication with the soft and by the way, OBS player do the same as VLC on my everyday pc.

**Comment by waldorf-statler** [comment, marked as answer]
On my side, the log seemed to refer to issues with the cache, after some checking decided best to re-install VLC and that did the trick. Thanks GerryDazoo for advice above, and actually thanks for making this possible! Now for trying to run the server in the cloud...

**Reply by Dhextreme** [reply]
Oh my god you found it. Same for me.
I uninstall and delete cache and preference, also deleted OBS and reinstall VLC,
it run very well now so far so good!!!

Very good decision waldorf-statler and thanks you for that one!

**Comment by Dhextreme** [comment]
I'll come later for Raspberry Pi that don't work and never gave a log file there /tmp/sling.log

Thanks for your inputs Guys and I tip my hat to you Gerry for a so great work!

**Reply by blasco403** [reply]
See post #164 for issues with starting on RP. Short version:

My RP had Python 2 and 3 installed, default was python2.

**Reply by Dhextreme** [reply]
Thank you blasco403. I found that last week and it work now. That was the solution.

**Comment by Pthepeg** [comment]
I have mine working on a pi.
First problem I ran into was in sling.service, ExecStart=/usr/bin/python3 -u slingbox_server.py
had to change it to python3 instead of python, to get it running.

I found the command line version of starting and stopping the slinger code was useful when first getting it going:
python3 slingbox_server.py config.ini     #to start, and ctrlC to stop.
